com.unity.multiplayer.mlapi/Runtime/Profiling/PerformanceDataManager.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace MLAPI.Profiling
+{
+    public static class PerformanceDataManager
+    {
+        static PerformanceTickData s_ProfilerData;
+        static int s_TickID;
+
+        public static void BeginNewTick()
+        {
+            s_ProfilerData = new PerformanceTickData
```

Comment:
I'm not sure how to solve, but this is going to allocate on the heap for every network tick. I could see re-using problematic since we shared this data out and who knows whether consumers hold a reference to it / process async. But I'm curious if you or any other reviewers have any thoughts on ways to avoid an alloc here

Reply:
If we absolutely have to do an allocation a pool might help

Reply:
+1 to Matt's point and also turning things into `struct`s as much as we can might help a little on GC pressure.

Reply:
also hashing and caching strings might help a little more on string allocation but I'm not sure how applicable these options are here.

Reply:
Since we went with the generic approach, it needs to exist on the heap (dictionary<,>) since it's an indeterminate size. Although we should definitely make the containers for the heap objects structs when possible to reduce it